python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1234,7 +1240,7 @@ def create_run(
             solids_to_execute=solids_to_execute,
             step_keys_to_execute=step_keys_to_execute,
             status=status,
-            tags=tags,
+            tags=dict(validated_tags),
```

Comment:
should `_construct_run_with_snapshots` take `frozentags` instead? guess this `dict` call is harmless

Reply:
I don't want to muck with the code that mangles the tags in _construct_run_with_snapshots in this PR.